code/modules/awaymissions/mission_code/Cabin.dm

```diff
@@ -19,56 +19,156 @@
 
 
 
+#define SECONDS_PER_LOG 150 //How many seconds a log will burn for
 
+//Fireplaces//
 
-/*Cabin code*/
 /obj/structure/fireplace
 	name = "fireplace"
+	desc = "A large stone fireplace, warm and cozy"
+	icon = 'icons/obj/fireplace.dmi'
+	icon_state = "fireplace"
+	density = 0
+	anchored = 1
+	pixel_x = -16
+	var/wood = 0
+	var/lit = 0
+
+
+/obj/structure/fireplace/proc/try_light(obj/item/O, mob/user)
+	if (!wood)
+		user << "<span class='warning'>[src] needs some wood to burn!</span>"
+		return FALSE
+	if (lit == 1)
+		user << "<span class='warning'>It's already lit!</span>"
+		return FALSE
+
+	var/lighting_text = "<span class='notice'>[user] lights the [src] with the [O].</span>"
+	if(istype(O, /obj/item/weapon/weldingtool))
```

Comment:
oh this is *JUST* for messages? that's even worse tbh, if it was for the functionality it'd be wrong still but this is just extra wrong.